### PR TITLE
[connector/datadog] Creating Datadog Connector To Replace Datadog Processor

### PR DESCRIPTION
**Pull request by gord02**
**Description:** <Describe what has changed.>

This pull request is for a newly created Datadog connector that replaces the need for the Datadog processor. This replacement is needed because the processor made use of a work-around that directly allowed the processor to export data. This is not something that adheres to the OpenTelemetry standard and thus the previous system should be deprecated and replaced by this new connector.

### CODE REVIEW

File: connector/datadogconnector/config.go [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package datadogconnector
+
+type Config struct {
```
Comment (songy23) [inline connector/datadogconnector/config.go:3]: Is this still WIP?

File: connector/datadogconnector/connector.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
+package datadogconnector
```
Comment (songy23) [inline connector/datadogconnector/connector.go:1]: Missing license headers.

**Review by songy23 — COMMENTED** [review]
Please add a README in datadogconnector and a changelog.

**Comment by atoulme** [comment]
Please open an issue for a new component and seek sponsors per instructions.

**Pull request closed without merging**